src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java

```diff
@@ -0,0 +1,93 @@
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalLocationBook(),
+            new UserPrefs(), getTypicalVisitBook());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), getTypicalLocationBook(),
+            new UserPrefs(), getTypicalVisitBook());
+
+    @Test
+    public void equals() {
+        Index firstIndex = Index.fromOneBased(1);
+        Index secondIndex = Index.fromOneBased(2);
+
+        GenerateLocationsCommand firstGenerateLocationsCommand = new GenerateLocationsCommand(firstIndex);
+        GenerateLocationsCommand secondGenerateLocationsCommand = new GenerateLocationsCommand(secondIndex);
+
+        // same object -> returns true
+        assertTrue(firstGenerateLocationsCommand.equals(firstGenerateLocationsCommand));
+
+        // same values -> returns true
+        GenerateLocationsCommand copyOfGenerateLocationsCommand = new GenerateLocationsCommand(firstIndex);
+        assertTrue(firstGenerateLocationsCommand.equals(copyOfGenerateLocationsCommand));
+
+        // different types -> returns false
+        assertFalse(firstGenerateLocationsCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(firstGenerateLocationsCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(firstGenerateLocationsCommand.equals(secondGenerateLocationsCommand));
+    }
+
+    @Test
+    public void execute_indexOutOfBounds_throwCommandException() {
+        String expectedMessage = MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+        Index index = Index.fromOneBased(100);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        assertThrows(CommandException.class, () -> command.execute(model));
+        try {
+            command.execute(model);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
+        }
+    }
+
+    @Test
+    public void execute_personAtIndexNotInfected_throwCommandException() {
+        String expectedMessage = MESSAGE_PERSON_IS_NOT_INFECTED;
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        assertThrows(CommandException.class, () -> command.execute(model));
+        try {
+            command.execute(model);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
+        }
+    }
+
+    @Test
+    public void execute_noVisitsFound_throwCommandException() {
+        String expectedMessage = MESSAGE_PERSON_HAS_NO_VISITS;
+        Index index = Index.fromOneBased(4);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        assertThrows(CommandException.class, () -> command.execute(model));
+        try {
+            command.execute(model);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
+        }
+    }
+}
```

Comment:
You have made tests that dealt with the edge cases. It would be good to have additional tests that prove the success of the command. In this case, it would be giving an infected person and confirming that the generated results are the expected results.